apps/etterlatte-behandling/src/main/kotlin/oppgave/OppgaveDao.kt

```diff
@@ -71,43 +71,34 @@ class OppgaveDao(private val connection: () -> Connection) {
                 |behandlingstype, soesken, b.prosesstype, adressebeskyttelse
                 |FROM behandling b INNER JOIN sak s ON b.sak_id = s.id 
                 |WHERE status = ANY(?) AND (b.prosesstype is NULL OR b.prosesstype != ?)
+                |AND ((adressebeskyttelse != ?) OR (adressebeskyttelse != ?)) 
                 """.trimMargin()
             )
             stmt.setArray(1, createArrayOf("text", statuser.toTypedArray()))
             stmt.setString(2, Prosesstype.AUTOMATISK.toString())
-            val oppgaver = stmt.executeQuery().toList {
-                val adressebeskyttelse = getString("adressebeskyttelse")
-                if (adressebeskyttelse != null &&
-                    (
-                        adressebeskyttelse == AdressebeskyttelseGradering.STRENGT_FORTROLIG.toString() ||
-                            adressebeskyttelse == AdressebeskyttelseGradering.STRENGT_FORTROLIG_UTLAND.toString()
-                        )
-                ) {
-                    null
```

Comment:
Funka veldig dårlig å gjøre sjekken her da dette fører til at alle resultater etter dette droppes.

Reply:
doc:
/**
 * Returns a sequence which invokes the function to calculate the next value on each iteration until the function returns `null`.
 *
 * The returned sequence is constrained to be iterated only once.
 *
 * @see constrainOnce
 * @see kotlin.sequences.sequence
 *
 * @sample samples.collections.Sequences.Building.generateSequence
 */
public fun <T : Any> generateSequence(nextFunction: () -> T?): Sequence<T> {
    return GeneratorSequence(nextFunction, { nextFunction() }).constrainOnce()
}